src/pages/TCY/components/Stake/Stake.tsx

```diff
@@ -84,13 +91,81 @@ export const StakeRoutes: React.FC<TCYRouteProps & { activeAccountNumber: number
   headerComponent,
   activeAccountNumber,
 }) => {
+  const translate = useTranslate()
   const location = useLocation()
   const [stakeTxid, setStakeTxid] = useState<string>()
   const queryClient = useQueryClient()
+  const dispatch = useAppDispatch()
+  const { getValues } = useFormContext<StakeFormValues>()
+  const { isDrawerOpen, openDrawer } = useActionCenterContext()
+
+  const toast = useToast({
+    duration: isDrawerOpen ? 5000 : null,
+    position: 'bottom-right',
+  })
+
+  const accountId = useAppSelector(state => {
+    const accountIdsByAccountNumberAndChainId = selectAccountIdByAccountNumberAndChainId(state)
+    const accountNumberAccounts = accountIdsByAccountNumberAndChainId[activeAccountNumber]
+    return accountNumberAccounts?.[thorchainChainId] ?? ''
+  })
 
   const handleTxConfirmed = useCallback(async () => {
+    if (!stakeTxid) throw new Error('Stake Txid is required')
+
+    const amountCryptoPrecision = bnOrZero(getValues('amountCryptoPrecision')).toFixed()
+
+    dispatch(
+      actionSlice.actions.upsertAction({
+        id: stakeTxid,
+        type: ActionType.GenericTransaction,
+        displayType: 'TCY',
```

Comment:
`displayType` brought as a notion here on top of type for generic transactions, since this is, well, a generic transaction notification, and each specific one will have their *actual* type we want for display only, e.g TCY here

Reply:
Nice catch ser! Agree re: consistent metadata pattern and enum, 6ac9599bf9